tests/e2e/simple_consensus.go

```diff
@@ -151,33 +178,36 @@ func SimpleConsensusTest(sc *RuntimeScenario, log *logging.Logger, conn *grpc.Cl
 		return fmt.Errorf("alice withdrawing with invalid nonce should fail")
 	}
 
+	log.Info("wait for changes to propagate to client node")
+	time.Sleep(propagationTime)
```

Comment:
is there a better way to do this?

and is it weird that this change pushed us over some cliff that we need this now?

Reply:
Yes for withdrawals, this should not be needed as you are already waiting on the consensus layer event. The problem is that you are trying to query consensus layer balances through the runtime which does not use the latest hegiht AFAIK. Why don't you query the consensus layer directly?

Reply:
> The problem is that you are trying to query consensus layer balances through the runtime which does not use the latest height AFAIK. Why don't you query the consensus layer directly?

oh correction on this: the consensus accounts balance query is not for obtaining information from the consensus _layer_. it gives the balance of the consensus _denomination_, still in the runtime layer

Reply:
Only deposits should require waiting for the next round, are you doing withdrawals and deposits at the same time? Of course withdrawals can in theory be refunded if they fail at the consensus layer, but that shouldn't happen.